pkg/build/controller/buildpod/controller.go

```diff
@@ -121,6 +121,9 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 	nextStatus := build.Status.Phase
 	currentReason := build.Status.Reason
 
+	// if the build is marked failed, the build status reason has already been
+	// set (probably by the build pod itself), so don't do any updating here
+	// or we'll overwrite the correct value.
```

Comment:
I don't anticipate that set changing.  If it does, that would be a good time to introduce a utility function but I don't think it needs to be done here/now.